examples/tic-tac-toe/cli/src/command.rs

```diff
@@ -0,0 +1,211 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::str::FromStr;
+
+use anyhow::{ensure, Context, Result};
+use clap::Parser;
+use sui_types::base_types::{ObjectID, SuiAddress};
+
+use crate::{
+    client::{Client, Connection},
+    crypto::public_key_from_base64,
+    game::GameKind,
+};
+
+#[derive(Parser, Debug)]
+pub enum Command {
+    /// Start a new game of tic-tac-toe.
+    New {
+        /// Use the shared version of the game (default).
+        #[clap(long, short)]
+        shared: bool,
+
+        /// Use the multi-sig version of the game.
+        #[clap(long, short)]
+        multi_sig: bool,
+
+        /// For a shared game, this is the opponent's address. For a multi-sig game, it is their
+        /// public key.
+        opponent: String,
+
+        #[clap(flatten)]
+        conn: Connection,
+    },
+
+    /// Make a move in an existing game.
+    Move {
+        /// ID of the game to make a move on.
+        game: ObjectID,
+
+        /// The row to place the move in.
+        #[clap(long, short)]
+        row: u8,
+
+        /// The column to place the move in.
+        #[clap(long, short)]
+        col: u8,
+
+        #[clap(flatten)]
+        conn: Connection,
+    },
+
+    /// Print the state of an existing game.
+    View {
+        /// ID of the game to view.
+        game: ObjectID,
+
+        #[clap(flatten)]
+        conn: Connection,
+    },
+
+    /// Delete a finished game.
+    Delete {
+        /// ID of the game to view.
+        game: ObjectID,
+
+        #[clap(flatten)]
+        conn: Connection,
+    },
+}
+
+impl Command {
+    /// Ensure the parameters for the command are valid.
+    fn validate(&mut self) -> Result<()> {
+        match self {
+            Command::New {
+                shared, multi_sig, ..
+            } => {
+                ensure!(
+                    !*shared || !*multi_sig,
+                    "Cannot specify both shared and multi-sig"
+                );
```

Comment:
I always find this macro hard hard to reason about.

Reply:
Probably might be better to use https://docs.rs/clap/latest/clap/struct.Arg.html#method.conflicts_with